From 189e7b939a87f6e591120940714140884b29454d Mon Sep 17 00:00:00 2001
From: "GONG, Ruiqi" <gongruiqi1@huawei.com>
Date: Tue, 12 Oct 2021 16:37:26 +0800
Subject: [PATCH] arm64: seccomp: fix compilation error with ILP32 support
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 189e7b939a87f6e591120940714140884b29454d


hulk inclusion
category: bugfix
bugzilla: 172971 https://gitee.com/openeuler/kernel/issues/I4DCXF

-------------------------------------------------

When ILP32 support is enabled, compilation on ARM64 would complain
about an undeclared __NR_compat_syscalls, since commit 0b18db62881c
("arm64: ilp32: introduce syscall table for ILP32") changes the config
that __NR_compat_syscalls depends on. Modify the config for
SECCOMP_ARCH_COMPAT_NR accordingly.

Fixes: 9571a6a35fdc ("arm64: Enable seccomp architecture tracking")
Signed-off-by: GONG, Ruiqi <gongruiqi1@huawei.com>
Reviewed-by: Chen Jiahao <chenjiahao16@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/include/asm/seccomp.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/include/asm/seccomp.h b/arch/arm64/include/asm/seccomp.h
index 22f74666e135..0be58ac682c4 100644
--- a/arch/arm64/include/asm/seccomp.h
+++ b/arch/arm64/include/asm/seccomp.h
@@ -52,10 +52,10 @@ static inline const int *get_compat_mode1_syscalls(void)
 #define SECCOMP_ARCH_NATIVE		AUDIT_ARCH_AARCH64
 #define SECCOMP_ARCH_NATIVE_NR		NR_syscalls
 #define SECCOMP_ARCH_NATIVE_NAME	"aarch64"
-#ifdef CONFIG_COMPAT
+#ifdef CONFIG_AARCH32_EL0
 # define SECCOMP_ARCH_COMPAT		AUDIT_ARCH_ARM
 # define SECCOMP_ARCH_COMPAT_NR	__NR_compat_syscalls
 # define SECCOMP_ARCH_COMPAT_NAME	"arm"
-#endif
+#endif /* CONFIG_AARCH32_EL0 */
 
 #endif /* _ASM_SECCOMP_H */
-- 
2.26.2

